internal/admission/validator_test.go

```diff
@@ -141,6 +154,58 @@ func TestKongHTTPValidator_ValidatePlugin(t *testing.T) {
 			wantMessage: ErrTextPluginConfigInvalid,
 			wantErr:     true,
 		},
+		{
+			name:      "plugin has valid configPatches",
+			PluginSvc: &fakePluginSvc{valid: true},
+			args: args{
+				plugin: kongv1.KongPlugin{
+					PluginName: "key-auth",
+					Config: apiextensionsv1.JSON{
+						Raw: []byte(`{"k1":"v1"}`),
+					},
+					ConfigPatches: []kongv1.ConfigPatch{
+						{
+							Path: "/foo",
+							ValueFrom: kongv1.ConfigSource{
+								SecretValue: kongv1.SecretValueFromSource{
+									Secret: "conf-secret",
+									Key:    "valid-conf",
+								},
+							},
+						},
+					},
+				},
+			},
+			wantOK:      true,
+			wantMessage: "",
+			wantErr:     false,
+		},
+		{
+			name:      "plugin has invalid configPatches",
+			PluginSvc: &fakePluginSvc{},
+			args: args{
+				plugin: kongv1.KongPlugin{
+					PluginName: "key-auth",
+					Config: apiextensionsv1.JSON{
+						Raw: []byte(`{"k1":"v1"}`),
+					},
+					ConfigPatches: []kongv1.ConfigPatch{
+						{
+							Path: "/foo",
+							ValueFrom: kongv1.ConfigSource{
+								SecretValue: kongv1.SecretValueFromSource{
+									Secret: "conf-secret",
+									Key:    "invalid-conf",
+								},
+							},
+						},
+					},
+				},
+			},
+			wantOK:      false,
+			wantMessage: ErrTextPluginConfigInvalid,
+			wantErr:     true,
```

Comment:
```suggestion
			wantMessage: ErrTextPluginConfigInvalid,
			wantErr:     false,
```
As far as I remember, a message is ignored when an error is returned and a 500 HTTP status code is returned with the error. We should either return:
- `ok = true, message = nil, err = nil` (when there's no validation errors)
- `ok = false, message = "validation rule failed", err = nil` (when there's a validation error detected)
- `ok = false, message = nil, err = "something went unexpectedly wrong - internal error"` (when something went wrong unexpectedly, e.g. we couldn't reach Kong).

Reply:
This sounds good, but this has changed the behavior of the original validator, so this change needs an entry in changelog (possibly in the `Fixed` section?)

Reply:
Updated, also updated `ValidateKongPlugin`.